src/animations/Rainbow.h

```diff
@@ -2,24 +2,25 @@
 
 class Rainbow : public Animation {
  private:
-  int initialPosition = 1;
-  int start = 1;
-  int finish = 255;
-  bool initialized = false;
-  bool loopRotary = true;
-  MyKnob &knob;
   CRGB *leds;
+  int32_t offset = 1;
+  KnobControl knobControl{1, 255, true};
 
-  void setup() {
-    knob.setDefaults(initialPosition, start, finish, loopRotary);
-  };
+ public:
+  Rainbow(CRGB leds_[]) : leds(leds_) {}
+
+  void wakeUp() {
```

Comment:
Why do we call these functions `wakeUp`?

Reply:
I changed this to `activate()`. Regardless, I was looking for a good name that wasn't `setup()`. I didn't want to reuse it because (AFAIK) the `setup()` method in `main.cpp` is only called once, while this is method is going to be called every time we go back to a particular animation. It's a minor distinction, but it made sense to me. I'm also open to calling it something else.

Reply:
Your thinking makes sense. I guess I was just following the existing pattern, but you're right, it's not an exact match. I do prefer `activate`.  👍